usr/sbin/pkcsslotd/log.c

```diff
@@ -268,11 +268,9 @@ BOOL NewLoggingFacility(char *ID, pLoggingFacility pStuff)
     pInfo->pid = 0;
     pInfo->LogLevel = pStuff->LogLevel;
 
-    sprintf(pInfo->Descrip, "%s %s", pStuff->Label, ID);
-
-    /* ensure that the last character is a NULL */
-    pInfo->Descrip[sizeof(pInfo->Descrip) - 1] = '\0';
-
+    /* truncate description to fit buffer */
+    snprintf(pInfo->Descrip, sizeof(pInfo->Descrip), "%s %s",
+             pStuff->Label, ID);
```

Comment:
Don't we need a zero termination here? What if snprintf ran to sizeof(pInfo->Descrip) and thus did not add the zero termination?

Reply:
snprintf always adds a 0-termination.  It copies at most n-1 chars and then places the 0 according to C standard.

Reply:
You are right, I was at strncpy, which does not....